source/jormungandr/jormungandr/interfaces/v1/serializer/JSONSchema.py

```diff
@@ -0,0 +1,165 @@
+# coding=utf-8
+
+# Copyright (c) 2001-2014, Canal TP and/or its affiliates. All rights reserved.
+#
+# This file is part of Navitia,
+#     the software to build cool stuff with public transport.
+#
+# Hope you'll enjoy and contribute to this project,
+#     powered by Canal TP (www.canaltp.fr).
+# Help us simplify mobility and open public transport:
+#     a non ending quest to the responsive locomotion way of traveling!
+#
+# LICENCE: This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Stay tuned using
+# twitter @navitia
+# IRC #navitia on freenode
+# https://groups.google.com/d/forum/navitia
+# www.navitia.io
+
+from serpy import fields, Serializer
+
+TYPE_MAP = {
+    'unknown': {
+        'type': 'unknown',
+    },
+    'method': {
+        'type': 'method',
+    },
+    'str': {
+        'type': 'string',
+    },
+    'float': {
+        'type': 'number'
+    },
+    'int': {
+        'type': 'integer'
+    },
+    'bool': {
+        'type': 'boolean',
+    },
+}
+
+class JSONSchema(Serializer):
+    properties = fields.MethodField('get_properties')
+    type = fields.MethodField('get_constant_object')
+    required = fields.MethodField('get_required')
+    # Weird name to ensure it will be processed a the end
+    _definitions = fields.MethodField('get_definitions', display_none=False, label='definitions')
```

Comment:
I don't really understand why is done, but I think we can do something else to not be dependant of the process order